src/main/java/org/minions/devfund/bruno/html/HtmlTag.java

```diff
@@ -0,0 +1,148 @@
+                && openTag != other.openTag;
+    }
+
+    /**
+     * Checks whether a tag is self-closing.
+     *
+     * @return if is self closing return true.
+     */
+    public boolean isSelfClosing() {
+        return SELF_CLOSING_TAGS.contains(element);
+    }
+
+    public boolean equals(Object obj) {
+        if (obj instanceof HtmlTag) {
+            HtmlTag other = (HtmlTag) obj;
+            return element.equals(other.element)
+                    && openTag == other.openTag;
+        }
+        return false;
+    }
+
+    public String toString() {
+        return "<".concat(openTag ? "" : "/")
+                .concat(element.equals("!--") ? "!-- --" : element).concat(">");
+    }
+
+    /**
+     * The remaining fields and functions are related to HTML file parsing.
+     */
+
+    // a set of tags that don't need to be matched (self-closing)
+    protected static final Set<String> SELF_CLOSING_TAGS = new HashSet<String>(
+            Arrays.asList("!doctype", "!--", "?xml", "xml", "area", "base",
+                    "basefont", "br", "col", "frame", "hr", "img",
+                    "input", "link", "meta", "param"));
+
+
+    protected static final String WHITESPACE = " \f\n\r\t";
+
+    public static Queue<HtmlTag> tokenize(String text) {
+        StringBuffer buf = new StringBuffer(text);
+        Queue<HtmlTag> queue = new LinkedList<HtmlTag>();
+
+        HtmlTag nextTag = nextTag(buf);
+        while (nextTag != null) {
+            queue.add(nextTag);
+            nextTag = nextTag(buf);
+        }
+        return queue;
+    }
+
+    protected static HtmlTag nextTag(StringBuffer buf) {
+        int openBracket = buf.indexOf("<");
+        int closeBracket = buf.indexOf(">");
+        if (openBracket >= 0 && closeBracket > openBracket) {
+            // check for HTML comments: <!-- -->
+            int commentIndex = openBracket + 4;
+            if (commentIndex <= buf.length()
+                    && buf.substring(openBracket + 1, commentIndex).equals("!--")) {
+                // look for closing comment tag -->
+                closeBracket = buf.indexOf("-->", commentIndex);
+                if (closeBracket < 0) {
+                    return null;
+                } else {
+                    buf.insert(commentIndex, " ");
+                    closeBracket += 3;    // advance to the closing bracket >
+                }
+            }
+
+            String element = buf.substring(openBracket + 1, closeBracket).trim();
+            // remove attributes
+            for (int i = 0; i < WHITESPACE.length(); i++) {
+                int attributeIndex = element.indexOf(WHITESPACE.charAt(i));
+                if (attributeIndex >= 0) {
+                    element = element.substring(0, attributeIndex);
+                }
+            }
+
+            // determine whether opening or closing tag
+            boolean isOpenTag = true;
+            int checkForClosing = element.indexOf("/");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Put single-quotes around '/' to use the faster "indexOf(char)" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3027)